Create 087-deprecating-the-old-docs-site.md (#387)
ipfs/blog: content/post/087-deprecating-the-old-docs-site.md

---
date: 2020-04-01
title: Deprecating the old docs site
author: Johnny Matthews
---

> # Deprecating the old docs site

Reviewer: ```suggestion
# Goodbye, IPFS legacy docs
```
... because it's a little snappier and it gets "IPFS" in there

The IPFS team is deprecating the old documentation website. That's just a fancy way of saying we're replacing the current docs.ipfs.io website with the new docs-beta.ipfs.io website.

## We built a new docs site

> Work on a new documentation site for IPFS started in Q3 of 2019. Before this, the docs site was built upon Hugo and hosted by Netlify. While Hugo is an excellent static-site generator, it lacks an encapsulated component model, relies upon old web development conventions, and requires DOM manipulation if you want to have any kind of site interactivity. Similarly, Netlify is an incredible hosting solution. However, the centralized nature of the service goes against our mission here at Protocol Labs.

Reviewer: The legacy docs actually weren't hosted on Netlify -- we were just using Netlify as a stand-in for hosting/deploying the docs beta while we wrapped up figuring out IPFS hosting. I'd just get rid of the whole Netlify part. Maybe add some thoughts around it just not being as easy to build on, particularly if not everyone in your org is conversant in Go. But in any case, I'd run all this past @cwaring for his blessing too.

Author: Ah ok, thought the legacy docs were still on Netlify. I'll take that part out then.

## The deprecation plan

We've built the site, put it online, and it's been well received. Next comes the tricky part: removing the old website and putting this one in its place.

Back in the early days of the internet, deprecating a site was as simple as putting up a notice saying things were going to be down for an hour or so, deleting everything in `/var/www/html`, and replacing it with the new website. Sadly the good-old-days are gone, and we now live in a world full of checks, balances, and redirects. With that in mind, here is our plan to deprecate the old site and replace it with the new one.

### Inform the world

The first step is to let everyone know what we're going to be doing. This post serves as part of the notification process, along with call-outs on the legacy site, and in the `ipfs/docs` GitHub repository.

### Plan for redirects

There have been several changes to the structure and naming conversions in the docs repo. Significant URL redirects must be put in place from legacy links to where the content now lives. The idea here is to minimize the amount of 404 errors encountered by the reader.

### Update external services

> Tools like Google Analytics and Bing Webmaster all need to be informed of the changes. Unfortunately, Alphabet's Google Search and Microsoft's bing are both necessary evils that enable us to reach as many users as possible. In the future, we'll be able to distance ourselves from centralized companies. Still, for now, we need to use them to help people access IPFS more easily.

Reviewer: ```suggestion
Tools like Google Analytics and Bing Webmaster all need to be informed of the changes. While they may not be major players in the decentralized web of tomorrow, for today at least, centralized search engines still are tools that help us reach as many users as possible and spread the word about IPFS and the dweb.
```
... just toned down the editorializing a little 😊

### Combine all the repos

> The IPFS docs have been stuck in limbo for a few months, with GitHub issues living in `ipfs/docs`, but the actual documentation and website content living in `ipfs/ipfs-docs-v2`. This has caused some confusion with posting issues in repo while reflecting those changes in another.

Reviewer: ```suggestion
The IPFS docs have been stuck in limbo for a few months, with GitHub issues living in `ipfs/docs`, but the actual documentation and website content living in `ipfs/ipfs-docs-v2`. This has caused some confusion with posting and commenting upon issues in one repo while reflecting those changes in another.
```
typo, clarity

> Past of the deprecation place is to merge `ipfs/ipfs-docs-v2` into `ipfs/docs`. All the content in `ipfs/docs` will be overwritten, but the issues will remain intact. Open issues will have their links checked to make sure they are referencing the correct and updated pages.

Reviewer: ```suggestion
Part of our deprecation plan is to merge the *code* in `ipfs/ipfs-docs-v2` into `ipfs/docs`. All the legacy docs site content in `ipfs/docs` will be overwritten, but the issues will remain intact. Open issues will have their links checked to make sure they are referencing the correct and updated pages.
```
typos, clarity

### Follow-up

> This final task will take place around 6-12 weeks after the site deprecation. There are three things the docs team will need to focus on at this stage:

Reviewer: ```suggestion
This final task will take place around 6-12 weeks after the site deprecation. There are four things the docs team will need to focus on at this stage:
```

1. Remove the _this URL has changed_ notices in any existing pages.
1. Re-crawl the site to make sure all the links and references are still working and up to date.

> 1. Check the top 404 referrers, and create a fix for them where possible.

Reviewer: ```suggestion
1. Check the top 404 referrers, and create a fix for them where possible.
2. Search all content in the IPFS and Protocol Labs GitHub orgs for references to old site pages, and fix those (so we don't 404 ourselves!).
```
Might want to call out this specific instance of fixing broken links, since it'll probably be the biggest offender.

---

> date: 2020-04-01

Reviewer 2: prooooobably good to avoid april fools day to any real announcements

Author: Haha! Yeah good point. Just threw that in there as a placeholder. I'll set it to Friday for now.

url: 2020-04-01-deprecating-the-old-docs-site
title: Deprecating the old docs site
author: Johnny Matthews
---

# Goodbye, IPFS legacy docs

> The IPFS team is deprecating the old documentation website. That’s just a fancy way of saying we’re replacing the current docs.ipfs.io website with the new docs-beta.ipfs.io website.

Reviewer 3: I would bump up the new docs hyperlink to the first mention of the new site.

## We built a new docs site

> Work on a new documentation site for IPFS started in Q3 of 2019. Before that, the docs site was built upon Hugo. While Hugo is an excellent static-site generator, it lacks an encapsulated component model, relies upon old web development conventions, and requires DOM manipulation if you want to have any kind of site interactivity.

Reviewer 3: Is any part of "relies upon old web development conventions" a bit too negative to mention them by name?

Author: Not that I know of, but I didn't want to dive too deep into this. There's already out in the wild about Hugo's issues.

Reviewer 2: I don't think these details are being helpful here. "encapsulated component model" / "old web developemnt conventions" / "DOM manipulation" are all jargon and not related to the user's actual question which is "what part of my docs usage experience will be better now?" Suggest we reframe around that context.

```suggestion
A common complaint from users new and old was that the IPFS onboarding experience was rough - and that the documentation was particularly difficult to navigate to find fast, reliable answers. We set out to change that! Work on a new documentation site for IPFS started in Q3 of 2019. Before that, the docs site was built upon Hugo. While Hugo is an excellent static-site generator, it was missing some features around interactive content that we needed to adaptively respond to user needs and collect feedback on which pages to improve first.
```

Author: Yeah that works better.

> So with all that in mind, the IPFS docs and UX working group built a new site to host the documentation. The site build was finished towards the end of 2019, and a beta version was made available to the public. Call-outs were placed on the current IPFS docs site to encourage readers to test the new website. You can see a preview of the site at [docs-beta.ipfs.io](https://docs-beta.ipfs.io).

Reviewer 2: ```suggestion
With that in mind, the IPFS docs and UX working group built a new site to host the documentation on Vuepress. After finishing the new site towards the end of 2019, we ran an opt-in public beta to get feedback and continue making improvements throughout Q1. You can see a preview of the site at [docs-beta.ipfs.io](https://docs-beta.ipfs.io).
```

Reviewer: VuePress (note intercap)

## The deprecation plan

We’ve built the site, put it online, and it’s been well received. Next comes the tricky part: removing the old website and putting this one in its place.

Back in the early days of the internet, deprecating a site was as simple as putting up a notice saying things were going to be down for an hour or so, deleting everything in `/var/www/html`, and replacing it with the new website. Sadly, the good-old-days are gone, and we now live in a world full of checks, balances, and redirects. With that in mind, here is our plan to deprecate the old site and replace it with the new one.

### Inform the world

The first step is to let everyone know what we’re going to be doing. This post serves as part of the notification process, along with call-outs on the legacy site, and in the `ipfs/docs` GitHub repository.

### Plan for redirects

> There have been several changes to the structure and naming conversions in the docs repo. Significant URL redirects must be put in place from legacy links to where the content now lives. The idea here is to minimize the amount of 404 errors encountered by the reader.

Reviewer 3: Conversions or conventions?

Author: Nice catch!

> There have been several changes to the structure and naming conventions in the docs repo. Significant URL redirects must be put in place from legacy links to where the content now lives. The idea here is to minimize the amount of 404 errors encountered by the reader.

Reviewer 2: We don't break links - that was one of the founding principles of IPFS. Can we do a full audit of all links that exist today (in either docs site) so we can ensure they're all mapped over to reasonable places in the new docs structure? (ex, heading links should redirect to the top-level article, where we've combined articles or broken them apart we should make a logical choice. We should NOT drop users on sad 404s)

Reviewer: We’ve already been doing this; note that pages on the legacy site all have callouts directing users to the _specific_ equivalent content, and vice versa.

Author: > implementing these equivalent-content links as redirects.

Yeah that's what I was going for. I'll change the last sentence to: _This is to ensure that nobody ends up on a `404` page, and is always directed to the content that they want to see._

### Update external services

> Tools like Google Analytics and Bing Webmaster all need to be informed of the changes. While they may not be major players in the decentralized web of tomorrow, for today at least, centralized search engines are still tools that help us reach as many users as possible and spread the word about IPFS and the dweb.

Reviewer 2: ```suggestion
Tools like Google Analytics and Bing Webmaster all need to be informed of the changes. While they may not be major players in the decentralized web of tomorrow, for today at least, centralized search engines are still tools that help us reach people with IPFS questions and spread the word about the dweb.
```

### Combine all the repos

The IPFS docs have been stuck in limbo for a few months, with GitHub issues living in `ipfs/docs`, but the actual documentation and website content living in `ipfs/ipfs-docs-v2`. This has caused some confusion with posting and commenting upon issues in one repo while reflecting those changes in another.

> Part of our deprecation plan is to merge the *code* in `ipfs/ipfs-docs-v2` into `ipfs/docs`. All the legacy docs site content in `ipfs/docs` will be overwritten, but the issues will remain intact. Open issues will have their links checked to make sure they are referencing the correct and updated pages.

Reviewer 2: ```suggestion
Part of our deprecation plan is to merge the *code* in `ipfs/ipfs-docs-v2` into `ipfs/docs`. All the legacy docs site content in `ipfs/docs` will be upgraded, but the issues will remain intact. Open issues will have their links checked to make sure they are referencing the correct and updated pages.
```

### Follow-up

This final task will take place around 6-12 weeks after the site deprecation. There are four things the docs team will need to focus on at this stage:

1. Remove the _this URL has changed_ notices in any existing pages.
2. Re-crawl the site to make sure all the links and references are still working and up to date.
3. Check the top 404 referrers, and create a fix for them where possible.

> 4. Search all content in the IPFS and Protocol Labs GitHub orgs for references to old site pages, and fix those (so we don’t 404 ourselves!).

Reviewer 2: I'm confused about which docs you are referencing here (links to the docs-v2? links to ipfs.docs?) and also why we expect to be hitting so many 404s in our existing docs/orgs

Author: > Re-crawl the site to make sure all the links and references are still working and up to date.

The new docs site. There's a plan to automate this step however, so this point isn't necessary.

> Check the top 404 referrers, and create a fix for them where possible.

If there are links coming in from external sites (other GitHub repos, tutorials, forum posts, etc) that are wrong, we can either redirect them to the homepage of docs, or redirect them to the most relevant page. Obviously the plan is to have no internal broken links.

## And that’s it

> After these steps are complete, the docs site will be fully ready to help readers explore the wonderful world of the decentralized web! These deprecation steps are set to begin in April and should be finished by June 2020. You can take a look at [the full plan over at hackmd.io](https://hackmd.io/@jessicaschilling/S1__jQ2TH).

Reviewer 2: lets make this a github issue in the docs repo - which takes the user to the right place to comment or add questions (And also see ongoing status!)

Reviewer: https://github.com/ipfs/docs/issues/475 has been open for a while, and contains a link to my original hackmd doc. Suggest linking to both of these, so readers can choose whether they’d prefer to follow the issue or just read the plan.

Author: My bad, should've linked to the issue and not the HackMD file.
